src/contract.rs

```diff
@@ -104,15 +134,27 @@ impl Auction {
         self.discount_rate = config.discount_rate;
         self.starting_price = config.starting_price;
 
-        msg::reply(
-            Event::AuctionStarted {
-                token_owner: self.nft.owner,
-                price: self.starting_price,
+        if let Err(e) = msg::send_for_reply(
+            self.nft.contract_id,
+            NFTAction::Transfer {
+                transaction_id,
+                to: exec::program_id(),
                 token_id: self.nft.token_id,
             },
             0,
         )
-        .unwrap();
+        .unwrap()
+        .await
+        {
+            gstd::debug!("{:?}", e);
+            return Err(Error::NftTransferFailed);
+        }
```

Comment:
I'd use `map_err` here to avoid `if let` with long condition.
```suggestion
        .map_err(|e| {
            gstd::debug!("{:?}", e);
            Error::NftTransferFailed
        })?;
```